docs/reference/rest-api/security/query-role.asciidoc

```diff
@@ -0,0 +1,285 @@
+[role="xpack"]
+[[security-api-query-role]]
+=== Query Role API
+
+++++
+<titleabbrev>Query Role</titleabbrev>
+++++
+
+Retrieves roles with <<query-dsl,Query DSL>> in a <<paginate-search-results,paginated>> fashion.
+
+[[security-api-query-role-request]]
+==== {api-request-title}
+
+`GET /_security/_query/role`
+
+`POST /_security/_query/role`
+
+[[security-api-query-role-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have at least the  `read_security` cluster privilege.
+
+[[security-api-query-role-desc]]
+==== {api-description-title}
+
+The role management APIs are generally the preferred way to manage roles, rather than using
+<<roles-management-file,file-based role management>>.
+The query roles API does not retrieve roles that are defined in roles files.
+You can optionally filter the results with a query. Also, the results can be paginated and sorted.
+
+[[security-api-query-role-request-body]]
+==== {api-request-body-title}
+
+You can specify the following parameters in the request body:
+
+`query`::
+(Optional, string) A <<query-dsl,query>> to filter which roles to return.
+The query supports a subset of query types, including
+<<query-dsl-match-all-query,`match_all`>>, <<query-dsl-bool-query,`bool`>>,
+<<query-dsl-term-query,`term`>>, <<query-dsl-terms-query,`terms`>>,
+<<query-dsl-match-query,`match`>>, <<query-dsl-ids-query,`ids`>>,
+<<query-dsl-prefix-query,`prefix`>>, <<query-dsl-wildcard-query,`wildcard`>>,
+<<query-dsl-exists-query,`exists`>>, <<query-dsl-range-query,`range`>>,
+and <<query-dsl-simple-query-string-query,`simple query string`>>.
++
+You can query the following values associated with a role.
++
+.Valid values for `query`
+[%collapsible%open]
+====
+`name`::
+The <<security-api-put-role-path-params,name>> of the role, which is internally indexed as a keyword.
```

Comment:
nit (optional): not sure if the ", which is internally indexed as a keyword." is needed. Maybe use the `(type)` prefix instead, so: 
```
(keyword) The <<security-api-put-role-path-params,name>> of the role
``` 

Same comment for all the fields.

Reply:
Good point, thanks!